### PR TITLE
change(consensus): Build Sprout and Sapling parameters into the zebrad binary, so a download server isn't needed

### LINKED ISSUE #6963: Integrate zk-SNARK parameters directly into the zebrad binary
The Sapling zk-SNARK parameters are small enough that it's sensible to store them in the `zebrad` binaries themselves so that users don't have to download and store them separately on disk, which is annoying. The Sprout (groth16) parameters are large, but only a minuscule portion of the Sprout parameters are actually needed by `zebrad` to verify Sprout transactions and these parameters can be stored directly in this repository and `include_bytes!`'d at compile time.

I wrote a [draft PR](https://github.com/zcash/zcash/pull/6661) for `zcashd` where we are considering doing the exact same thing. The PR uses the `wagyu-zcash-parameters` Rust crate so that Sapling parameters are effectively downloaded by `cargo` at compile time. Also in this proposed change for `zcashd`, people who need to create Sprout proofs will have to independently download the Sprout proving key like before, but by default this is not required. `zebrad` would need no such compatibility hack, because as far as I know Sprout proving is not supported nor will it be in the future.

Is this something the project would be interested in adding? If so I might be able to throw a PR up here along the lines of my PR in `zcashd`.

### LINKED ISSUE #7613: feature: Compile sprout and sapling parameters into the Zebra binary
### Motivation

The `zcashd` team have recently started compiling the Sprout and Sapling parameters into their binary:
> What we are now doing is we have bundled the Sprout verification and the sapling proving and verification parameters into the Zcashd binary.
> 
> Whereas now the largest component of these is the sapling spend parameters, which are about 50 megabytes.
> 
> If you take the existing release binaries and add about 16 megabytes, that's going to be approximately the right ballpark.

https://github.com/ZcashCommunityGrants/arboretum-notes/blob/main/AllArboristCallNotes/Arborist%20Call%2061-Notes.md#1-zebrad-status---zebra-update

We've had issues with the parameter download server being unreliable or rate-limited before. They are likely to get worse now that ECC is no longer using it on a regular basis.

### Specifications

This `bundled-prover` feature allows access to bundled Sapling parameters:
https://docs.rs/zcash_proofs/latest/zcash_proofs/prover/struct.LocalTxProver.html#method.bundled

I'm not sure if there is a Rust crate for bundled Sprout verification parameters.

### Complex Code or Requirements

This might make our CI builds slower to link, or use more disk space.

### Testing

Our existing unit tests cover this code, but we might want to check that our integration tests have run on Sprout and Sapling transactions. We can do that by checking the block heights from the cached state to the tip had those transactions.

Sapling is common, but Sprout is rare now.

### Related Work

We might need to do this as after the zcashd 5.7.0 release, depending on ECC's API deprecations and download server support guarantees.

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

We don't want to rely on a single external download server when Zebra first runs.

Close #7613
Close #6963 

Fixes known issue in README with parameter downloads
Reduces the size of the full Zebra install and cache on disk by 700 MB (we don't need the full Sprout parameters any more)

This PR series will eventually:
- Simplify CI and makes it more reliable
- Remove unnecessary code and tests from Zebra

### Specifications

This PR is based on the corresponding `zcashd` PR https://github.com/zcash/zcash/pull/6661

### Complex Code or Requirements

Loading bytes from inside the binary is much simpler, as long as our build disks are large enough. (The final size is smaller because we drop the sprout prover and transcript from the build and binaries.)

## Solution

Production code:
- load sapling parameters from inside the binary
- load sprout parameters from inside the binary
- remove parameter download code and dependency features
- reword log messages and documentation (partial)

CI and testing:
- remove download examples
- remove workflow steps that download parameters

Documentation:
- add a changelog entry based on `zebrad`'s changelog entry

### Testing

We have existing comprehensive tests for the Sprout and Sapling parameters. Sapling is tested regularly by our integration and sync tests.

Sprout might not be tested, so 

## Review

This is a routine change, but it is blocking cleanup work in other PRs.

### Reviewer Checklist

  - [ ] Will the PR name make sense to users?
    - [x] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
    - [x] Does it change concurrent code, ~unsafe code,~ or consensus rules?
  - [ ] How do you know it works? Does it have tests?

Please manually check that the new sprout file contains the correct data. Here is how the `zcashd` team did it:
https://github.com/zcash/zcash/pull/6661/files#r1248159359

### CODE REVIEW

**Comment by teor2345** [comment]
`rpc_endpoint_parallel_threads` failed with a timing issue, we might want to fix that test to use the logs rather than a fixed time.

https://github.com/ZcashFoundation/zebra/actions/runs/6607410396/job/17944979809?pr=7800#step:4:6196

This is unrelated and will be cleared the next time I push.

File: zebra-consensus/src/primitives/groth16/params.rs [hunk 1]
```diff
@@ -55,164 +34,54 @@ pub struct SaplingParameters {
...
 
         Groth16Parameters { sapling, sprout }
     }
 
-    /// Returns the path to the Groth16 parameters directory.
-    pub fn directory() -> PathBuf {
-        zcash_proofs::default_params_folder().expect("unable to find user home directory")
-    }
-
     /// Returns a hint that helps users recover from parameter download failures.
```
Comment (upbqdn) [inline zebra-consensus/src/primitives/groth16/params.rs]:
```suggestion
    /// Returns a hint that helps users recover from parameter loading failures.
```

**Review by upbqdn — DISMISSED** [review]

File: zebra-consensus/src/primitives/groth16/params.rs (hunk 1 above)
Comment (teor2345) [reply zebra-consensus/src/primitives/groth16/params.rs]: Thanks!

**Review by upbqdn — DISMISSED** [review]

File: .github/workflows/ci-coverage.yml [hunk 2]
```diff
@@ -52,8 +57,7 @@ jobs:
     name: Coverage on stable
     # The large timeout is to accommodate:
     # - stable builds (typically 30-50 minutes), and
```
Comment (teor2345) [inline .github/workflows/ci-coverage.yml]:
```suggestion
    # - stable builds (typically 50-90 minutes), and
```

File: .github/workflows/ci-coverage.yml [hunk 3]
```diff
@@ -52,8 +57,7 @@ jobs:
...
-    # - parameter downloads (an extra 90 minutes, but only when the cache expires)
-    timeout-minutes: 140
+    timeout-minutes: 60
```
Comment (teor2345) [inline .github/workflows/ci-coverage.yml]:
```suggestion
    timeout-minutes: 120
```

File: .github/workflows/ci-unit-tests-os.yml [hunk 4]
```diff
@@ -68,8 +68,7 @@ jobs:
     name: Test ${{ matrix.rust }} on ${{ matrix.os }}${{ matrix.features }}
     # The large timeout is to accommodate:
     # - macOS and Windows builds (90 minutes, typically 30-70 minutes), and
```
Comment (teor2345) [inline .github/workflows/ci-unit-tests-os.yml]:
```suggestion
    # - macOS and Windows builds (typically 50-90 minutes), and
```

File: .github/workflows/ci-unit-tests-os.yml [hunk 5]
```diff
@@ -68,8 +68,7 @@ jobs:
...
-    # - parameter downloads (an extra 100 minutes, but only when the cache expires)
-    timeout-minutes: 190
+    timeout-minutes: 90
```
Comment (teor2345) [inline .github/workflows/ci-unit-tests-os.yml]:
```suggestion
    timeout-minutes: 120
```

File: zebra-consensus/src/primitives/groth16/params.rs [hunk 6]
```diff
@@ -55,164 +34,54 @@ pub struct SaplingParameters {
...
-        let sapling_spend_path = params_directory.join(zcash_proofs::SAPLING_SPEND_NAME);
-        let sapling_output_path = params_directory.join(zcash_proofs::SAPLING_OUTPUT_NAME);
-        let sprout_path = params_directory.join(zcash_proofs::SPROUT_NAME);
+        tracing::info!("checking and loading Zcash Sapling and Sprout parameters");
 
-        Groth16Parameters::download_sapling_parameters(&sapling_spend_path, &sapling_output_path);
-        Groth16Parameters::download_sprout_parameters(&sprout_path);
+        let (sapling_spend_bytes, sapling_output_bytes) =
+            wagyu_zcash_parameters::load_sapling_parameters();
+        let sprout_vk_bytes = include_bytes!("sprout-groth16.vk");
```
Comment (teor2345) [inline zebra-consensus/src/primitives/groth16/params.rs:54]: The content of this file needs to be manually checked by a reviewer before merging.

See the `zcashd` PR for details:
https://github.com/zcash/zcash/pull/6661/files#r1248159359

Comment (upbqdn) [reply zebra-consensus/src/primitives/groth16/params.rs:54]: I confirm the first 1828 bytes of `sprout-groth16.vk` and `sprout-groth16.params` are identical:

```bash
cmp -n 1828 sprout-groth16.vk ~/.zcash-params/sprout-groth16.params
```

**Review by upbqdn — APPROVED** [review]

**Pull request merged**